packages/relay/tests/lib/services/cacheService/cacheService.spec.ts

```diff
@@ -244,27 +244,30 @@ describe('CacheService Test Suite', async function () {
       key2: 'value2',
       key3: 'value3',
     };
-    let redisInMemoryServer: RedisInMemoryServer;
 
-    this.beforeAll(async () => {
-      redisInMemoryServer = new RedisInMemoryServer(logger.child({ name: `in-memory redis server` }), 6381);
-      await redisInMemoryServer.start();
+    useInMemoryRedisServer(logger, 6381);
 
-      process.env.REDIS_ENABLED = 'true';
-      process.env.REDIS_URL = 'redis://127.0.0.1:6381';
-      process.env.TEST = 'false';
-      process.env.MULTI_SET = 'true';
-      cacheService = new CacheService(logger.child({ name: 'cache-service' }), registry);
-    });
+    let multiSet: string | undefined;
+    let disconnected: boolean;
 
-    this.afterAll(async () => {
-      await redisInMemoryServer.stop();
-      process.env.TEST = 'true';
+    const disconnectRedisClient = async () => {
       await cacheService.disconnectRedisClient();
-    });
+      disconnected = true;
+    };
 
     this.beforeEach(async () => {
-      await cacheService.clear();
+      multiSet = process.env.MULTI_SET;
+      process.env.MULTI_SET = 'true';
+      cacheService = new CacheService(logger.child({ name: 'cache-service' }), registry);
+      disconnected = false;
```

Comment:
Would it be worth exploring adding some state to the cache service, such as `isConnected`, number of connections, etc, instead of tracking in the test like this?  Asking because we once ran into an issue with the relay and the number of connections to the shared cache.

Reply:
Done, added new methods for:
- `isConnected()`
- `getNumberOfConnections()`